solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseCloudSolrClient.java

```diff
@@ -476,8 +479,9 @@ public void registerDocCollectionWatcher(String collection, DocCollectionWatcher
     assertZKStateProvider().zkStateReader.registerDocCollectionWatcher(collection, watcher);
   }
 
-  @SuppressWarnings({"unchecked"})
-  private NamedList<Object> directUpdate(AbstractUpdateRequest request, String collection) throws SolrServerException {
+  private CompletableFuture<NamedList<Object>> directUpdate(AbstractUpdateRequest request,
```

Comment:
I have to wonder if it's worthwhile to have directUpdate support Http2 Async callback given the complexity here?  I *do* like it internally for parallel updates -- doUpdatesWithExecutor.